ui/pages/notifications/notifications-list-item.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import type { Notification } from '../../../app/scripts/controllers/metamask-notifications/types/notification/notification';
+import { Box } from '../../components/component-library';
+import {
+  BlockSize,
+  Display,
+  FlexDirection,
+} from '../../helpers/constants/design-system';
+import { NOTIFICATIONS_ROUTE } from '../../helpers/constants/routes';
+import { useMarkNotificationAsRead } from '../../hooks/metamask-notifications/useNotifications';
+import { useMetamaskNotificationsContext } from '../../contexts/metamask-notifications/metamask-notifications';
+import {
+  NotificationComponents,
+  hasNotificationComponents,
+} from './notification-components';
+
+export function NotificationsListItem({
+  notification,
+}: {
+  notification: Notification;
+}) {
+  const history = useHistory();
+  const { listNotifications } = useMetamaskNotificationsContext();
+
+  const { markNotificationAsRead } = useMarkNotificationAsRead();
+
+  const handleNotificationClick = useCallback(() => {
+    markNotificationAsRead([
+      {
+        id: notification.id,
+        type: notification.type,
+        isRead: notification.isRead,
+      },
+    ]);
+    listNotifications();
```

Comment:
Why do we refetch the list?

Reply:
Instead of refetching, lets instead add logic to the `markAsRead` where it calls our server AND also updates local cache.

Future ticket

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-655